Tests/MongoSwiftTests/SpecTestRunner/Match.swift

```diff
@@ -0,0 +1,95 @@
+import Foundation
+import MongoSwift
+import Nimble
+
+// swiftlint:disable line_length
+/// Protocol that allows a type to assert it matches a given value according to the specs' MATCHES function.
+/// https://github.com/mongodb/specifications/tree/master/source/connection-monitoring-and-pooling/tests#spec-test-match-function
+internal protocol Matchable {
+    /// Returns whether this MATCHES the expected value according to the function defined in the spec.
+    /// https://github.com/mongodb/specifications/tree/master/source/connection-monitoring-and-pooling/tests#spec-test-match-function
+    func matches(expected: Any) -> Bool
+}
+// swiftlint:enable line_length
+
+/// Extension that adds MATCHES functionality to `Array`.
+extension Array: Matchable {
+    internal func matches(expected: Any) -> Bool {
+        guard !isPlaceholder(expected) else {
+            return true
+        }
+
+        guard let expected = expected as? [Any], expected.count <= self.count else {
```

Comment:
why do we check for `<=` rather than `==`? it seems like we would incorrectly match `[1,2]` and `[1,2,3]` this way. the way that the match function is written does seem to omit comparing lengths. 
it makes more sense to me to do that in the document case, since maybe new fields get added to a command or something, and they wouldn't be in the spec test data but would show up in the actual values. 
but I am not sure the same really applies to arrays.... is it safe to assume any extra values at the end of the actual array don't matter?

Reply:
The pseudocode for MATCH has the following loop for arrays:
```
If expected is a JSON array:
    For every idx/value in expected:
      Assert that actual[idx] MATCHES value
```
So we only ever loop over elements of `expected`, meaning extra stuff at the end of `actual` is okay. An important thing to note is that MATCH is not a symmetric function (i.e. a MATCHES b does not imply b MATCHES a).

Reply:
yeah, i read that, i guess my question was more like: why wouldn't we care about the extra stuff? 
thinking back to that mongoc thing with the extra events emitted in pooled mode, it feels like not explicitly comparing lengths could hide spec violations where extra events are published or extra commands are sent. 
but i guess this is a question for the author of MATCH, not you 🙂

Reply:
Yeah...I'm not sure, maybe for consistency with how document is treated or just for ease of test writing? I do know that the change stream tests were written with this definition in mind, seeing as the command event arrays for each test case are only subsets of the actual events fired over the course of each case.